include/onnxruntime/core/framework/stream_handles.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <functional>
+#include <unordered_map>
+#include "core/framework/ortdevice.h"
+
+namespace onnxruntime {
+class IExecutionProvider;
+// this opaque handle could be anything the target device generated.
+// it could be a cuda event, or a cpu notification implementation
+using NotificationHandle = void*;
+// it can be either a cuda stream, or even nullptr for device doesn't have stream support like cpu.
+using StreamHandle = void*;
+
+// a stream abstraction which hold an opaque handle, and a reference to which EP instance this stream belong to.
```

Comment:
1. How is EP instance captured here given that Stream contains OrtDevice?
2. The class needs more documentation as it's not clear why timestamp, the unordered_map, etc are needed here.

Reply:
yes the comment is out of date, we initially start group stream with EP instance, but later we found we need group with device to enable case like TRT/Cuda share the same stream